Remove the use of deprecated tranport methods in tests

unkcpz opened the pull request.

unkcpz commented:
When running tests, I got a lot resource not clean warnings which may indicate some resource leak:

```
tests/transports/test_all_plugins.py::test_copy[core.local]
  /home/jyu/venv/aiida-core-dev/lib/python3.12/site-packages/psycopg/_connection_base.py:146: ResourceWarning: connection <psycopg.Connection [IDLE] (host=localhost port=46513 user=guest database=01029ab2-f1cd-4c66-a7f1-a405af5e5324) at 0x775bc419b920> was deleted while still open. Please use
 'with' or '.close()' to close the connection
    warn(
```

This require some investigation.

khsrali commented on tests/transports/test_all_plugins.py:
```diff
@@ -33,16 +30,22 @@
     elif request.param == 'core.ssh_auto':
         kwargs = {'machine': 'localhost'}
-        filepath_config = tmp_path / 'config'
+        filepath_config = tmp_path_factory.mktemp('transport') / 'config'
```
Maybe add a comment for one line, why you did it this way..
probably because `listdir` would also list `'config'` for `core.auto_ssh`

khsrali commented on tests/transports/test_all_plugins.py:
```diff
@@ -365,278 +297,244 @@ def test_dir_reading_permissions(custom_transport):
-        transport.putfile(local_file_name, remote_file_name)
-        transport.getfile(remote_file_name, retrieved_file_name)
+        transport.putfile(local_file_abs_path, remote_file_abs_path)
+        transport.getfile(remote_file_abs_path, retrieved_file_abs_path)
```
I know it wasn't done properly, but perhaps we leave it like this so that both `getfile` & `get` somehow would get tested?

khsrali commented on tests/transports/test_all_plugins.py:
```diff
@@ -365,278 +297,244 @@ def test_dir_reading_permissions(custom_transport):
-    local_dir = os.path.join('/', 'tmp')
-    remote_dir = local_dir
+    local_dir = tmp_path_factory.mktemp('local')
+    remote_dir = tmp_path_factory.mktemp('remote')
```
to be consistence, can you also define `local_tmp_path` fixture? 
Because now, in some tests only use `remote_tmp_path`, some other directly uses `tmp_path_factory` ..
Either that, or I suggest just remove that `remote_tmp_path` fixture and always to use `tmp_path`

khsrali commented on tests/transports/test_all_plugins.py:
```diff
@@ -365,278 +297,244 @@ def test_dir_reading_permissions(custom_transport):
-        transport.remove(remote_file_name)
-        # If it couldn't end the copy, it leaves what he did on
-        # local file
-        assert 'file_retrieved.txt' in transport.listdir('.')
```
why this `assert` is removed?

unkcpz replied:
Right, I added it back.

khsrali replied:
Thanks! let's put it back before the second `getfile`.
As the first one supposed to already download it, and second one it appears to re-download with `overwrite=True`
 (weird test, but anyways)

khsrali commented on tests/transports/test_all_plugins.py:
```diff
@@ -710,251 +608,238 @@ def test_put_and_get_overwrite(
+
+        # third test put. Can copy one file into a new file
+        transport.put(str(local_base_dir / '*.tmp'), str(remote_workdir / 'prova'))
+        assert transport.isfile(str(remote_workdir / 'prova'))
```
probably this is not quite the same as  the former line(?)
```python
assert set(['prova', 'local']) == set(transport.listdir('.'))
```

unkcpz replied:
The assert I added is more strict and I think want required to be tested. 
The `put` is to move a file to `prova`, the previous test only test the "moved" thing appear in the new dst folder. If a "prova" folder exist, then the previous test still passed but it is not we want. 
The "local" is a side effect of using the same folder to test things. It is not expect to be exist in the remote.

khsrali replied:
ah ok, I was confused with `local`.. all good!

khsrali commented on tests/transports/test_all_plugins.py:
```diff
@@ -710,251 +608,238 @@ def test_put_and_get_overwrite(
         # i.e., the destination should be a folder
         with pytest.raises(OSError):
-            transport.put(os.path.join(local_base_dir, '*.txt'), 'prova')
-        # copy of folder into file
```
not clear why these lines are removed?
```python
        # copy of folder into file

        with open(os.path.join(local_dir, directory, 'existing.txt'), 'w', encoding='utf8') as fhandle:
            fhandle.write(text)
        with pytest.raises(OSError):
            transport.put(os.path.join(local_base_dir), 'existing.txt')
        transport.remove('existing.txt')
```

unkcpz replied:
Delete by mistake, I add it back as:

```python
        # can't copy folder to an exist file
        with open(remote_workdir / 'existing.txt', 'w', encoding='utf8') as fhandle:
            fhandle.write(text)
        with pytest.raises(OSError):
            transport.put(str(local_base_dir), str(remote_workdir / 'existing.txt'))
        transport.remove(str(remote_workdir / 'existing.txt'))
```

Nice catch!!

khsrali commented on tests/transports/test_all_plugins.py:
```diff
@@ -1047,73 +916,50 @@ def test_put_get_empty_string_tree(custom_transport):
-        with open(filepath, 'wb') as handle:
-            handle.write(content)
+    with custom_transport as transport:
+        transport.gettree(str(remote_tmp_path / 'sub' / 'path'), str(tmp_path / 'sub' / 'path'))
```
Here I think we are supposed to check the functionality of `gettree`. Only this would be serve the purpose of the test. I understand, it was incompetent even before your changes :)
```suggestion
        transport.gettree(str(remote_tmp_path), str(tmp_path))
```

unkcpz replied:
cool! I didn't realize that, the tests passed with your change!

khsrali commented on tests/transports/test_all_plugins.py:
```diff
@@ -1047,73 +916,50 @@ def test_put_get_empty_string_tree(custom_transport):
 
 
-def test_exec_pwd(custom_transport):
+def test_exec_pwd(custom_transport, remote_tmp_path):
```
I would keep the previous `pwd` and instead make it exclusive to `core.ssh` &  `core.auto_ssh` &  `core.local`.
This test would not make sense for plugins without `getcwd`

khsrali commented on tests/transports/test_all_plugins.py:
```diff
@@ -1209,25 +1055,19 @@ def test_transfer_big_stdout(custom_transport):
 
-    with custom_transport as trans:
+    with custom_transport as transport:
+        transport: Transport
```
This line is a bit out of blue (?) is not practiced in the similar places..
```suggestion
```

unkcpz replied:
Right, I was putting it here only for letting the LSP help me to autocomplate on methods. Removed.

khsrali reviewed (changes requested):
Hi @unkcpz ,
Thanks a lot! I think this PR is useful.

unkcpz commented:
This PR require the change in https://github.com/aiidateam/aiida-core/pull/6639 to not fail the mypy.

unkcpz commented on tests/transports/test_all_plugins.py:
```diff
@@ -62,26 +65,36 @@ def test_is_open(custom_transport):
-def test_makedirs(custom_transport):
-    """Verify the functioning of makedirs command"""
+def test_deprecate_chdir_and_getcwd(custom_transport, remote_tmp_path):
+    """Test to be deprecated ``chdir``/``getcwd`` methods still work."""
```
But at the moment, all three transport all has these methods, no?

khsrali replied:
fair enough!

unkcpz reviewed (commented):
Thanks @khsrali  for review! I think I address all your comments.

khsrali commented on tests/transports/test_all_plugins.py:
```diff
@@ -33,16 +30,29 @@
     elif request.param == 'core.ssh_auto':
         kwargs = {'machine': 'localhost'}
-        filepath_config = tmp_path / 'config'
+        # The transport config is store in a indepenednt tmp path to not mix up with the files under operating.
```
just to make it more clear:
```suggestion
        # Please note, `tmp_path/transport/config' keeps existing during all the following tests!
```

unkcpz replied:
```suggestion
        # The transport config is store in a independent temporary path per test to not mix up with the files under operating.
```

I did a test on printing the config path for every test, it is actually for every test the transport / 'config' is under the folder of its own as `/tmp/pytest-of-jyu/pytest-24/transport7/config` with the number after "transport" incremented. 

I adapt the comment string a bit, hope it is better.

khsrali commented on tests/transports/test_all_plugins.py:
```diff
@@ -365,278 +301,242 @@ def test_dir_reading_permissions(custom_transport):
         # TODO : get doesn't retrieve empty files.
         # Is it what we want?
-        transport.get(remote_file_name, retrieved_file_name)
+        transport.get(remote_file_abs_path, retrieved_file_abs_path)
```
```suggestion
        transport.get(remote_file_abs_path, retrieved_file_abs_path)
        assert Path(retrieved_file_abs_path).exists()
```

khsrali commented on tests/transports/test_all_plugins.py:
```diff
@@ -365,278 +301,242 @@ def test_dir_reading_permissions(custom_transport):
-        assert 'file_retrieved.txt' in transport.listdir('.')
-        os.remove(retrieved_file_name)
+        transport.getfile(remote_file_abs_path, retrieved_file_abs_path)
+        assert 'file_retrieved.txt' in [p.name for p in (local_dir / directory).iterdir()]
```
```suggestion
```

unkcpz replied:
I change it to:

```python
        transport.get(remote_file_abs_path, retrieved_file_abs_path)
        assert Path(retrieved_file_abs_path).exists()
        t1 = Path(retrieved_file_abs_path).stat().st_mtime_ns


        # overwrite retrieved_file_name in 0.01 s
        time.sleep(0.01)
        transport.getfile(remote_file_abs_path, retrieved_file_abs_path)
        assert Path(retrieved_file_abs_path).exists()
        t2 = Path(retrieved_file_abs_path).stat().st_mtime_ns

        # Check st_mtime_ns to sure it is override
        assert t2 > t1
```

To make sure it is override by the second operation of `getfile`.

khsrali reviewed (commented):
Thanks @unkcpz , just a few more adjustments.. then I think would be good to go

unkcpz commented:
Thanks @khsrali, I addressed the comments you mentioned, and improve a test to make sure the `getfile` is override the retrieved_file download by `get` method.

khsrali reviewed (approved):
Thanks a lot @unkcpz , LGTM!
